blocks/gmo-program-details/gmo-program-details.js

```diff
@@ -14,12 +14,47 @@ const platformMappings = getMappingArray('platforms');
 export default async function decorate(block) {
     const encodedSemi = encodeURIComponent(';');
     const encodedProgram = encodeURIComponent(programName);
+
+    blockConfig = readBlockConfig(block);
+
     const programQueryString = `getProgramDetails${encodedSemi}programName=${encodedProgram}${encodedSemi}programID=${encodeURIComponent(programID)}`;
+    const deliverableQueryString = `getProgramDeliverables${encodedSemi}programName=${encodedProgram}${encodedSemi}programID=${encodeURIComponent(programID)}`;
+
+    // Immediately render a placeholder header
+    block.innerHTML = `
+    <div class="back-button">
+        <span class="icon icon-back"></span>
+        <span class="back-label">Back</span>
+    </div>
+    <div class="main-body-wrapper">
+        <div class="placeholder-header">Loading program details...</div>
+    </div>
+    `;
+
+    // Wait for program data to render the actual header
     const programData = await executeQuery(programQueryString);
     const program = programData.data.programList.items[0];
-    blockConfig = readBlockConfig(block);
     const header = buildHeader(program, queryVars).outerHTML;
-    if (!program) {
+
+    // Update the header with the actual data
+    block.querySelector('.placeholder-header').outerHTML = header;
+
+    let imageObject = null;
+    let totalassets = 0;
```

Comment:
Is 0 better than null for a default value here? If an error occurs, is the expected behaviour to display a 0?

Reply:
cc: @TyroneAEM

Reply:
Since the imageObject is expected to have the following JSON object
e.g {imageUrl : '', imageAltText: '', assetCount: 0}

I will change 
let imageObject = null
to
let imageObject = {imageUrl : '', imageAltText: '', assetCount: 0};